node/core/pvf/src/host.rs

```diff
@@ -33,6 +33,7 @@ use futures::{
 	Future, FutureExt, SinkExt, StreamExt,
 };
 use polkadot_parachain::primitives::ValidationResult;
+use polkadot_primitives::vstaging::ExecutorParams;
```

Comment:
This comment pertains to the whole PVF host design.

One thing that I don't see here is that there is no separation between the artifacts. As you noticed, you parametrize the compilation process with the semantics structure. This is not a fluke, it was designed that way since different semantics could lead to different artifacts.

Take for example `canonicalize_nans`. If wasmtime, or to be specific cranelift, is configured with that option, during machine code generation it will insert special snippets that ensure that NaNs are cannonical. IOW, 

```
prepare(wasm, semantics { canonicalize_nans: true }) ≠ prepare(wasm, semantics { canonicalize_nans: false })
```

In the current implementation (which I admittedly only glanced over) it seems to not account for that fact, and could potentially reuse an artifact.